.github/workflows/check-md-links.yml

```diff
@@ -0,0 +1,23 @@
+name: Check Markdown links
+# https://github.com/gaurav-nelson/github-action-markdown-link-check
+
+on:
+  workflow_call:
+    push:
+      branches: [main]
+    pull_request:
+      branches: [main]
+      types: [opened, reopened, ready_for_review, synchronize]
+      paths:
+        - ".github/workflows/md-check-links.yml"
+        - "**/*.md"
+
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@master
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        use-quiet-mode: 'yes'
+        config-file: '.github/workflows/markdown.links.config.json'
```

Comment:
if we g with `workflow_call` then this shall be input argument
also I would consider to make this file on the fly (dump to the CI run) if none is provided :chipmunk:

Reply:
That makes sense, did the first part. Will have to think about how to do the second bit. Based of initial ideas (ignore my github workflow building skills), would the jobs part of the workflow look like this,
```
jobs:
  markdown-link-check:
    runs-on: ubuntu-20.04
    steps:    
    - uses: actions/checkout@master
    - name: Check for config file, if not create it
      run: |
        if [ ! -f ${{ inputs.config-file }} ]; then
          echo "config.json not found, creating one..."
          echo '{
              "ignorePatterns": [
                {
                  "pattern": "^https://github.com/Lightning-AI/lightning/pull/.*"
                }
              ],
              "httpHeaders": [
                {
                  "urls": ["https://github.com/", "https://guides.github.com/", "https://help.github.com/", "https://docs.github.com/"],
                  "headers": {
                    "Accept-Encoding": "zstd, br, gzip, deflate"
                  }
                }
              ]
            }' > ${{ inputs.config-file }}
        else
          echo "config.json found, no need to create one."
        fi
    - uses: gaurav-nelson/github-action-markdown-link-check@v1
      with:
        use-quiet-mode: 'yes'
        config-file: ${{ inputs.config-file }}
```